storage/innobase/buf/buf0dblwr.cc

```diff
@@ -1231,7 +1278,18 @@ buf_dblwr_flush_buffered_writes(
 
 		/* Check that the page as written to the doublewrite
 		buffer has sane LSN values. */
-		buf_dblwr_check_page_lsn(write_buf + len2);
+		buf_dblwr_check_page_lsn(dblwr_page);
+
+		/* Disable encryption of Page 0 of any tablespace or if it is
+		system tablespace, do not encrypt pages upto TRX_SYS_PAGE_NO
+	        (including). TRX_SYS_PAGE should be not encrypted because dblwr
+		buffer is found from this page */
+		if (encrypt_parallel_dblwr
+		    && block->page.id.page_no() != 0
```

Comment:
Extract this logic to a helper function (w/o encrypt_parallel_dblwr) and reuse below too

Reply:
Fixed